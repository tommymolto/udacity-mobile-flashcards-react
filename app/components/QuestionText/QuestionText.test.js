/* global describe, it, expect */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import QuestionText from './QuestionText';
import storeConfig from '../../store/storeConfig';

describe('QuestionText', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={storeConfig}>
        <QuestionText
          deckTitle="React"
          currentQuestionNum={1}
          question="question"
          quizLength={3}
          onPress={() => {}}
        />
              </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
