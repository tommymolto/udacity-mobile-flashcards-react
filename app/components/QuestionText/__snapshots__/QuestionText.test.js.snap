// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionText renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 20,
          "paddingHorizontal": 15,
          "paddingVertical": 25,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    question
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "#05AE54",
            "color": "#ffffff",
            "fontSize": 22,
            "height": 45,
            "marginLeft": 40,
            "marginRight": 40,
            "marginVertical": 10,
            "paddingBottom": 10,
            "paddingTop": 10,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Show answer
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 12,
          "fontStyle": "italic",
          "paddingHorizontal": 15,
          "paddingVertical": 25,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    2
     question(s) remaining including this one
  </Text>
</View>
`;
